services/repository/contributors_graph.go

```diff
@@ -58,8 +58,9 @@ func GetContributorStats(ctx context.Context, cache cache.Cache, repo *repo_mode
 		_, run := generateLock.Load(cacheKey)
 		if run {
 			return nil, ErrAwaitGeneration
+		} else {
+			generateLock.Store(cacheKey, struct{}{})
 		}
-		generateLock.Store(cacheKey, struct{}{})
```

Comment:
@6543 I reverted commits until 8c35b11fd and used status code 202. While reverting the commits, these changes of yours were reverted too. Were they important? Were they bugs that you fixed?